sphinx/parsers.py

```diff
@@ -55,15 +55,23 @@ def set_application(self, app):
         self.info = app.info
 
 
-class RSTParser(docutils.parsers.rst.Parser):
+class RSTParser(docutils.parsers.rst.Parser, Parser):
     """A reST parser customized for Sphinx."""
 
+    def set_application(self, app):
+        # type: (Sphinx) -> None
+        Parser.set_application(self, app)
+        if hasattr(SphinxSmartQuotes, 'smartquotes_action'):
+            SphinxSmartQuotes.smartquotes_action = self.env.settings['smartquotes_action']
```

Comment:
I have a problem here which is that afaik SphinxSmartQuotes is a class, not an instance of that class, and in get_transforms below it is the class object which is added to list of transforms. Hence I needed to modify the class attribute here. Don't know if this can have undesired consequences.

Reply:
Right, class attribute is a kind of global variable. So modifying it is not good way. So it would be better to update it temporarily like following:
```
from contextlib import contextmanager

@contextmanager
def sphinx_smartquotes_action(env):
    try:
        original = SphinxSmartQuotes.smartquotes_action
        SphinxSmartQuotes.smart_quotes_action = env.config.smartquotes_action
        yield
     finally:
         SphinxSmartQuotes.smart_quotes_action = original

# in env.doc_read() method:
with sphinx_smartquotes_action(self):
    doctree = read_doc(...)
```

Reply:
Thanks, will do.